superset/dashboards/schemas.py

```diff
@@ -245,6 +245,7 @@ class DashboardDatasetSchema(Schema):
     time_grain_sqla = fields.List(fields.List(fields.Str()))
     granularity_sqla = fields.List(fields.List(fields.Str()))
     normalize_columns = fields.Bool()
+    time_secondary_column = fields.Bool()
```

Comment:
Since this is a bool, a better name might be `always_filter_primary_temporal_column`, though that's huge. Maybe `filter_primary_time_column`?

Reply:
`always_filter_main_dttm`?